cmake/templates/HPXConfig.cmake.in

```diff
@@ -44,8 +42,24 @@ set(HPX_Fortran_COMPILER "@CMAKE_Fortran_COMPILER@" CACHE STRING "Fortran compil
 set(HPX_Fortran_COMPILER_ID "@CMAKE_Fortran_COMPILER_ID@" CACHE STRING "Fortran compiler id for HPX" FORCE)
 set(HPX_Fortran_COMPILER_VERSION "@CMAKE_Fortran_COMPILER_VERSION@" CACHE STRING "Fortran compiler version for HPX" FORCE)
 
+##################################################
+# Setup the imported libraries (publicly linked) #
+set(HPX_PROGRAM_OPTIONS_WITH_BOOST_PROGRAM_OPTIONS_COMPATIBILITY
+  @HPX_PROGRAM_OPTIONS_WITH_BOOST_PROGRAM_OPTIONS_COMPATIBILITY@
+  CACHE STRING "" FORCE)
+set(HPX_FILESYSTEM_WITH_BOOST_FILESYSTEM_COMPATIBILITY
+  @HPX_FILESYSTEM_WITH_BOOST_FILESYSTEM_COMPATIBILITY@
+  CACHE STRING "" FORCE)
+set(HPX_WITH_CXX17_FILESYSTEM @HPX_WITH_CXX17_FILESYSTEM@)
+# Separate targets to be unarily linked to some modules
 set(HPX_BOOST_ROOT "@BOOST_ROOT@")
```

Comment:
```suggestion
set(BOOST_ROOT "@BOOST_ROOT@")
```?

Reply:
It's actually a test, I suspect it can work (to set anything, `HPX_BOOST_ROOT` being just a information to the user) because of the `find_dependency` macro, but let's see..

Reply:
So `BOOST_ROOT` is required to actually find Boost. I suggest we always set `HPX_BOOST_ROOT`, and then, *if* `BOOST_ROOT` (or `BOOST_INCLUDEDIR`, `BOOST_LIBRARYDIR`) is not set, we set `BOOST_ROOT` to `HPX_BOOST_ROOT` as a fallback. Then, if the user sets a different `BOOST_ROOT` they will first be warned and if they really know what they're doing they can set `HPX_IGNORE_BOOST_COMPATIBILITY=ON`. But by default we would just use the same `BOOST_ROOT` that was used for building HPX. Same would go for hwloc except that it doesn't have a compatibility check (maybe it needs one as well?).